### DISCUSSION TITLE
Does kyuubi support spark sql, flink sql, hive sql on k8s?

### DISCUSSION THREAD

**Discussion by kissycn** [apache/kyuubi#4785] [Q&A] [answered]
Does kyuubi support spark sql, flink sql, hive sql on k8s?

**Comment by pan3793** [comment]
That depends on

1. whether the computing engine itself supports running on K8s
2. whether Kyuubi supports launching the computing engine in K8s mode

Specific to the engines you listed

- [x] Spark
- [ ] Flink (2 is not implemented)
- [ ] Hive (neither 1 nor 2 is sufficient)

**Comment by kissycn** [comment]
First, thanks for your reply my bro
2、For flink, does the community have a plan for Kyuubi supports launching the computing engine in K8s mode?
I don’t know the implementation details of kyuubi, so I don’t know whether the implementation difficulty of flink sql on k8s is high or not. Is it feasible for me or other community members to contribute to the difficulty?
3、For hive, hive supports spark as the execution engine, so theoretically hive sql on spark can run on k8s'

**Reply by pan3793** [reply]
@link3280 may have ideas for 2

for 3, the engine of Kyuubi is almost a HiveServer2 instead of MR/Spark/Tez

**Reply by pan3793** [reply]
@iodone I remember that HiveServer on YARN/K8s cases mentioned in your previous share, do you have more details for that?

**Reply by link3280** [reply, marked as answer]
@kissycn 

> 2、For flink, does the community have a plan for Kyuubi supports launching the computing engine in K8s mode?

I think so. We're doing it step by step.

> Is it feasible for me or other community members to contribute to the difficulty?

Definitely, contributions are welcome. You could start by supporting `kubernetes-session` mode in Flink local engine if you are interested.

**Reply by kissycn** [reply]
ok i'll try it
